### PR TITLE
STARK: Fix for the cursor over border issue #1401

### PR DESCRIPTION
**Pull request by mj0331**
Removing the coordinate bounds will make the cursor show up over the black border, as requested in issue #1401 , however it's possible to interact with elements 'hidden' by the border(for example moving the character 'offscreen'). A possible improvement/fix would be to have a bounded cursor position for game logic and a unbounded position for rendering

### CODE REVIEW

**Comment by mj0331** [comment]
As a sidenote, I know it would have been better to start a new branch instead of working on master, although I was a bit pressured by time. I'll start new branches in the future

File: engines/stark/ui/cursor.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ void Cursor::setCursorImage(VisualImageXMG *image) {
 
 
 void Cursor::setMousePosition(const Common::Point &pos) {
-	_mousePos = _gfx->getScreenPosBounded(pos);
```
Comment (bgK) [inline engines/stark/ui/cursor.cpp]: Ok, now `getScreenPosBounded` is not used anymore and can be removed.

Also this does not make sure the cursor is set to inactive when over the black borders. For example, in the TLJ demo, when hovering the quit button in the top bar and moving the mouse to the right over the black border, the cursor stays active.

Comment (mj0331) [reply engines/stark/ui/cursor.cpp]: Would adding a function `Gfx::Driver::isPosInScreenBounds` be fine? I could probably use that to do some checks for borders, maybe in `Stark::Cursor::Render`

Comment (bgK) [reply engines/stark/ui/cursor.cpp]: Ok, sounds fine. Did you notice `Driver::getScreenViewport`?

Comment (mj0331) [reply engines/stark/ui/cursor.cpp]: Yeah, shall I use that for bound checking?

Comment (bgK) [reply engines/stark/ui/cursor.cpp]: You might want to.

**Comment by mj0331** [comment]
I think I need to scale the cursor's position when the game is in fullscreen, because just a direct `_screenViewport.contains(pos)` doesn't seem to cut it. I'm not 100% sure what and how to scale it so I'm comparing things in the same coord space

**Comment by bgK** [comment]
Cursor::render already manipulates both original and native coordinates. Does it not work when making the test using native coordinates?

**Comment by mj0331** [comment]
With both native and original coordinates, the bound test fails. My render function looks like this, but it thinks the cursor is out of bounds when it's on the left half of the screen:
https://gist.github.com/mj0331/1f61944594da69542693d0baeb837c84

**Comment by bgK** [comment]
`isPosInScreenBounds` expects native coordinates while you are passing it original resolution coordinates. Using native mouse coordinates should work:
```cpp
    if (!_gfx->isPosInScreenBounds(_mousePos)) {
        setCursorType(Cursor::kPassive);
    }
```

**Comment by mj0331** [comment]
Yeah, I tried with native coords as well, and the bound check still fails when the cursor is in the left quarter of the screen. I'll need to debug this a little further

**Comment by mj0331** [comment]
Alright, this should be it

**Comment by bgK** [comment]
> Alright, this should be it

Yes, looks ok. One last little thing. Notice how your code uses spaces for intentation while the rest uses tabs. Please change that.
(These are the code formatting conventions we follow: http://wiki.scummvm.org/index.php/Code_Formatting_Conventions)

**Comment by mj0331** [comment]
Oops, I thought CLion converted those into tabs but I was wrong. My bad

**Comment by bgK** [comment]
All good, thanks!

**Pull request merged**